### PR TITLE
Navbar logout

### PR DESCRIPTION
**Pull request by damfinkel**
## Summary

Dropdown from user avatar in nav var. Profile redirects to home for the time being, and the logout option logs out and redirects to the login screen.
The dropdown dismisses by touching outside it or touching the profile button again.
The WBooks logo redirects to home.

## Screencast

df-training

### CODE REVIEW

File: src/components/Dropdown.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import './Dropdown.css';
+
+class Dropdown extends React.Component {
+  componentWillMount = () => document.addEventListener('click', this.handleClick, false);
```
Comment (sbalay) [inline src/components/Dropdown.js]: you can remove the listener on willMount and willUnmount if we are using the ones on willReceiveFront

File: src/components/NavigationBar.js [hunk 2]
```diff
@@ -1,24 +1,103 @@
...
-        <img src={wbooksLogo} alt="WBooks" className="navigation-logo" />
-      </div>
-      <div className="nav-bar-buttons-container">
-        <button src={notificationsSvg} alt="Notifications" className="nav-button nav-notifications-button" />
-        <button src={addBookSvg} alt="Add Book" className="nav-button nav-add-book-button" />
-        <button src={notificationsSvg} alt="Settings" className="nav-button nav-avatar-button" />
+import Dropdown from './Dropdown';
+import './NavigationBar.css';
+
+const axiosInstance = axios.create({
```
Comment (sbalay) [inline src/components/NavigationBar.js]: use the instance from `config/api.js`

File: src/components/NavigationBar.js [hunk 3]
```diff
@@ -1,24 +1,103 @@
...
+  baseURL: 'https://wbooks-api-stage.herokuapp.com/api/v1/'
+});
+
+const logout = (onSuccess, onError) => {
```
Comment (sbalay) [inline src/components/NavigationBar.js]: move all the interactions with the autentication api to a service. we usually create services in `src/services` and divide them using models logic. i.e: AuthService, BooksService, CommentsService, etc.

**Review by sbalay — CHANGES_REQUESTED** [review]

File: src/components/NavigationBar.js (hunk 2 above)
Comment (damfinkel) [reply src/components/NavigationBar.js]: it doesn't exist in this branch, I refactored this later on. I'd normally add a new card for this but there's no trello haha

File: src/components/NavigationBar.js [hunk 4]
```diff
@@ -1,24 +1,96 @@
...
-        <img src={wbooksLogo} alt="WBooks" className="navigation-logo" />
-      </div>
-      <div className="nav-bar-buttons-container">
-        <button src={notificationsSvg} alt="Notifications" className="nav-button nav-notifications-button" />
-        <button src={addBookSvg} alt="Add Book" className="nav-button nav-add-book-button" />
-        <button src={notificationsSvg} alt="Settings" className="nav-button nav-avatar-button" />
+import Dropdown from './Dropdown';
+import './NavigationBar.css';
+
+const logout = (onSuccess, onError) => {
```
Comment (sbalay) [inline src/components/NavigationBar.js:12]: This is duplicated here and in authservice. it should be in authservice

**Review by sbalay — APPROVED** [review]

**Pull request closed without merging**